RoboJackets/apiary
# Show payments history on user profile page

* Adds payment history on the User Profile page. Both dues and travel payments will appear in the list
  * As much transaction payment detail as available will be displayed
  * Added backend HTTP Resources for Travel and TravelAssignment resources to achieve this
  * $0 transactions without card details are omitted from the list (these generally come from incomplete Square transactions)
* Added the fancy centered loading component to the OAuth2 client / PAT tables

Test instructions:
* Go to https://apiary-test.robojackets.org/profile and look at your payments list
  * It would also be good to fabricate some transactions like refunds and/or impersonate a few other users to spot-check how it looks to them

TODO:
- [x] Change to order payments in descending `updated_at` order
- [x] Add min width on payment table columns to improve UI on small screens
- [x] Unit tests for payments API endpoint (dues transactions)
- [x] Unit tests incorporating travel payments
- [ ] Verify payments history for users in apiary-test
- [x] Fix https://robojackets.sentry.io/issues/3920778010

**Screenshots**
* Kitchen sink of transaction types
  * ![image](https://user-images.githubusercontent.com/5790137/216883981-99993d0b-82e7-419d-a2b5-ba771066e2cb.png)
* Empty state
  * ![image](https://user-images.githubusercontent.com/5790137/216883649-fe939794-9043-41dc-8fb2-7b0398ab33b4.png)
* Loading state
  * ![image](https://user-images.githubusercontent.com/5790137/216883255-7ed15371-9a60-40ac-bfd8-c3aff256221b.png)
* Error state
  * ![image](https://user-images.githubusercontent.com/5790137/216883198-d7e06321-c48b-4cc1-a9a6-6177936c5bb7.png)

Closes #228
